source/rpc/proxy.py

```diff
@@ -1,23 +1,41 @@
+import socket
 import xmlrpc.client
 from contextlib import contextmanager
 from threading import Timer
-from .session import Session
-from .edit import Edit
-from .application import Application
-from .imager import Imager
+# from .session import Session
+# from .edit import Edit
+# from .application import Application
+# from .imager import Imager
+
+SOCKET_TIMEOUT = 10
 
 
 class BaseProxy(object):
     """Base class for all proxies."""
 
-    def __init__(self, url):
+    def __init__(self, url, timeout=SOCKET_TIMEOUT):
         """Initialize the actual xmlrpc.client.ServerProxy from given url.
 
         Args:
             url (str): url for xmlrpc.client.ServerProxy
         """
-        self.__transport = xmlrpc.client.Transport()
-        self.__proxy = xmlrpc.client.ServerProxy(uri=url, transport=self.__transport)
+        try:
+            socket.setdefaulttimeout(timeout)
```

Comment:
This set sets default globally! Not only for rpc, but also for new PCIC and for every other socket that will be made after that.

Reply:
You are right.
Modified code to first initializing a socket transport connection with timeout value and passing this one to the proxy. Test are passing all after this change.

```
    def __init__(self, address, url, timeout=SOCKET_TIMEOUT):
        """Initialize the actual xmlrpc.client.ServerProxy from given url.

        Args:
            address (str): ip address of host
            url (str): url for xmlrpc.client.ServerProxy
            timeout (float): argument can be a non-negative floating point number
            expressing seconds, or None. If None, SOCKET_TIMEOUT value is used as default
        """
        try:
            self.__transport = xmlrpc.client.Transport()
            self.__transport.make_connection(host=address)
            getattr(self.__transport, "_connection")[1].timeout = timeout
            self.__proxy = xmlrpc.client.ServerProxy(uri=url, transport=self.__transport)
        except TimeoutError:
            self.close()
```

Reply:
fixed in 4d73fe6aa170398e8adc34b0aaca5b33eb6746c1